### PR TITLE
[LEADS-415] Responses streaming support

### PR DESCRIPTION
**Pull request by xmican10**
## Description

Followup PR with supporting responses `stream=True` parameter

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up service version
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Unit tests improvement

## Tools used to create PR

Identify any AI code assistants used in this PR (for transparency and review context)

- Assisted-by: (e.g., Claude, CodeRabbit, Ollama, etc., N/A if not used)
- Generated by: (e.g., tool name and version; N/A if not used)

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

## Summary by CodeRabbit

* **New Features**
  * Added OpenAI-style streaming support to the `/responses` endpoint for real-time delivery
  * Improved streaming parsing, including tool-call extraction and file-search chunk mapping into the response payload
  * Enhanced token timing metrics for better generation performance tracking
* **Bug Fixes**
  * Improved handling of tool-call argument shapes and stricter validation for missing/invalid fields
* **Tests**
  * Extended unit tests to verify correct streaming vs non-streaming request behavior and end-to-end streaming parsing (including error and `[DONE]` edge cases)

### CODE REVIEW

**Comment by asamal4** [comment]
@coderabbitai full review

**Comment by xmican10** [comment]
I need to rebase..

**Comment by xmican10** [comment]
@coderabbitai full review

File: src/lightspeed_evaluation/core/api/streaming_parser.py [hunk 1]
```diff
@@ -69,53 +87,181 @@ def to_response_dict(self) -> dict[str, Any]:
...
+    item_handlers = {
+        "mcp_call": _responses_handle_mcp_call_item,
+        "file_search_call": _responses_handle_file_search_item,
+    }
+    handler = item_handlers.get(item.get("type", ""))
+    if handler:
+        handler(ctx, item)
+
+
+def _responses_handle_mcp_call_item(
```
Comment (asamal4) [inline src/lightspeed_evaluation/core/api/streaming_parser.py]: didn't we add similar logic in _parse_mcp_call_item

File: src/lightspeed_evaluation/core/api/streaming_parser.py [hunk 2]
```diff
@@ -204,22 +346,19 @@ def _parse_tool_call(event_data: dict[str, Any]) -> Optional[dict[str, Any]]:
         Dictionary with tool_name, arguments, and optional result, or None if invalid.
     """
     try:
-        # Support both "name"/"args" (new format) and "tool_name"/"arguments" (legacy)
```
Comment (asamal4) [inline src/lightspeed_evaluation/core/api/streaming_parser.py:207]: Why are we removing these comments ?

File: tests/unit/core/api/test_streaming_parser.py [hunk 3]
```diff
@@ -136,21 +138,21 @@ def test_parse_response_with_new_format_tool_calls(
         lines = [
             'data: {"event": "start", "data": {"conversation_id": "conv_new"}}',
             'data: {"event": "tool_call", "data": '
-            '{"id": "tc_1", "name": "pods_list", "args": {"namespace": "default"}}}',
+            '{"id": "tc_1", "name": "get_weather", "args": {"city": "London"}}}',
```
Comment (asamal4) [inline tests/unit/core/api/test_streaming_parser.py]: Why are we changing this ? This is just test data - but still I would prefer to keep it more realistic + there is no technical reason for chaning this.

```suggestion
            '{"id": "tc_1", "name": "pods_list", "args": {"namespace": "default"}}}',
```

**Review by asamal4 — COMMENTED** [review]
Thanks !! Some minor comments - applicable in multiples places

File: src/lightspeed_evaluation/core/api/streaming_parser.py (hunk 1 above)
Comment (xmican10) [reply src/lightspeed_evaluation/core/api/streaming_parser.py]: yes, it makes sense to merge it

File: src/lightspeed_evaluation/core/api/streaming_parser.py (hunk 2 above)
Comment (xmican10) [reply src/lightspeed_evaluation/core/api/streaming_parser.py:207]: i missed that, i'm adding them back

File: tests/unit/core/api/test_streaming_parser.py (hunk 3 above)
Comment (xmican10) [reply tests/unit/core/api/test_streaming_parser.py]: I added new tests with dummy data so i thought it would be better to unify the theme across the test file. But I agree with you, I'll reverse these changes.

**Review by asamal4 — APPROVED** [review]
